.github/workflows/update-os-snapshots.yml

```diff
@@ -0,0 +1,250 @@
+name: Refresh E2E Snapshots
```

Comment:
Those are some complicated config files, with cryptic values... 😩

Reply:
I agree that it's quite complicated and much love is required to get it more readable.
For now, I'm going on the 'just make it work' (that's why it's a copy of the CI workflow)

I want to keep the scope tight and not affect build.yml too much, but I planned an issue to refactor our workflow to match the quality of our more recent projects.

The overall complexity is overall inherent to the nature of what we test, which is essentially a native application, with bindings deep in the browser OS.


TL;DR: The overall complexity is not really shrinkable. It can be made more manageable by refactoring and splitting it into reusable (and digestible) actions. However, I keep that for another PR